mne/preprocessing/maxwell.py

```diff
@@ -505,3 +537,107 @@ def _update_sss_info(raw, origin, int_order, ext_order, nsens):
         raw.info['proc_history'] = [proc_block]
 
     return raw
+
+
+def _read_fine_cal(info, fine_cal_fname):
+    """Read sensor locations and imbalances from fine calibration file."""
+
+    logger.info('Fine calibration file used in Maxwell Filter: ' +
+                fine_cal_fname)
+
+    # Read new sensor locations
+    with open(fine_cal_fname, 'r') as fid:
+        cal_chans = []
+        lines = fid.readlines()
+        lines = lines[::-1]
+        while len(lines) > 0:
+            line = lines.pop()
+            if line[0] != '#':
+                vals = np.fromstring(line, sep=' ')
+
+                # Check for correct number of items
+                if len(vals) not in [14, 16]:
+                    raise RuntimeError('Error reading fine calibration file')
+
+                ch = int(vals[0])
+                ch_name = 'MEG' + '%04d' % ch  # Zero-pad names to 4 characters
+                loc = vals[1:13]
+                imbalance = vals[13:]
```

Comment:
FYI this will keep all of the original `vals` arrays in memory, because slices are always copies, and thus `vals` cannot be cleared from memory. It's fine here because there are so few so we don't need to worry about it, but something to keep in mind for future work. If you really want to be clean, just add `.copy()` to the end of these, and the garbage collector will do its work.